### ISSUE TITLE
Logic error in `ConfigParser::readConfig`

### ISSUE THREAD

**Issue by youlanhai** [cocos2d/cocos2d-x#15571]
The temporary search path was inserted into the begin of the array, but the last element was removed from the array when operation finished.

https://github.com/cocos2d/cocos2d-x/blob/v3/tools/simulator/libsimulator/lib/runtime/ConfigParser.cpp

``` c++
void ConfigParser::readConfig(const string &filepath)
{
    ...
    // add writable path to search path temporarily for reading config file
    vector<std::string> searchPathArray = FileUtils::getInstance()->getSearchPaths();
    searchPathArray.insert(searchPathArray.begin(), FileServer::getShareInstance()->getWritePath());
    FileUtils::getInstance()->setSearchPaths(searchPathArray);
    ...
    // revert search path
    searchPathArray.erase(searchPathArray.end() - 1);
    FileUtils::getInstance()->setSearchPaths(searchPathArray);
    ...
}
```

Notice, the code `searchPathArray.erase(searchPathArray.end() - 1);` means to erase the last element.

**Comment by minggo** [comment]
Yep, it is a bug.

**Comment by minggo** [comment]
fixed in #15581
